pkg/handler/rpms.go

```diff
@@ -26,11 +26,46 @@ func RegisterRpmRoutes(engine *echo.Group, rDao *dao.DaoRegistry) {
 	addRepoRoute(engine, http.MethodGet, "/snapshots/:uuid/rpms", rh.listSnapshotRpm, rbac.RbacVerbRead)
 	addRepoRoute(engine, http.MethodGet, "/snapshots/:uuid/errata", rh.listSnapshotErrata, rbac.RbacVerbRead)
 	addRepoRoute(engine, http.MethodPost, "/snapshots/rpms/names", rh.searchSnapshotRPMs, rbac.RbacVerbRead)
+	addRepoRoute(engine, http.MethodPost, "/snapshots/module_streams/search", rh.searchSnapshotModuleStreams, rbac.RbacVerbRead)
 	addRepoRoute(engine, http.MethodPost, "/rpms/presence", rh.detectRpmsPresence, rbac.RbacVerbRead)
 	addTemplateRoute(engine, http.MethodGet, "/templates/:uuid/rpms", rh.listTemplateRpm, rbac.RbacVerbRead)
 	addTemplateRoute(engine, http.MethodGet, "/templates/:uuid/errata", rh.listTemplateErrata, rbac.RbacVerbRead)
 }
 
+// searchSnapshotModuleStreams godoc
+// @Summary      List modules and their streams for snapshots
+// @ID           searchSnapshotModuleStreams
+// @Description  List modules and their streams for snapshots
+// @Tags         snapshots
+// @Accept       json
+// @Produce      json
+// @Param        body  body   api.SearchModuleStreamsRequest  true  "request body"
+// @Param  	     uuids	path	[]string	true	"Snapshot IDs."
+// @Param  	     package_names	path	[]string	true	"Package Names"
+// @Success      200   {object}  api.SearchModuleStreamsCollectionResponse
+// @Failure      400 {object} ce.ErrorResponse
+// @Failure      401 {object} ce.ErrorResponse
+// @Failure      404 {object} ce.ErrorResponse
+// @Failure      500 {object} ce.ErrorResponse
+// @Router       /snapshots/module_streams/search [post]
+func (rh *RpmHandler) searchSnapshotModuleStreams(c echo.Context) error {
```

Comment:
I'd put this in a new handler file for snapshots, not in the rpm one

Reply:
I was wondering about this, as I saw "listSnapshotErrata" in the same file.

Reply:
yep, probably

Reply:
although maybe @rverdile has some opinions as we aren't super consistent about this

Reply:
In rpms.go

<img width="1020" alt="Screenshot 2024-12-03 at 9 02 27 AM" src="https://github.com/user-attachments/assets/37852c81-2231-4a07-96e0-8ec0c5938de5">

In snapshots.go

<img width="1034" alt="Screenshot 2024-12-03 at 9 03 40 AM" src="https://github.com/user-attachments/assets/556e0adc-7efd-499e-8fd7-e6443d8e5606">

Reply:
rpms.go so far has been the place we put things related to searching rpm content units, so my instinct would have been to put it there as well

Reply:
maybe it'd be less confusing to just organize these things based on the URL path and not usage, but that's a bigger change

Reply:
i was okay with leaving the handler here, but then i saw that the dao method is also being added to the rpm doa.  Which its just one function, but i'm about to add a bunch more as part of  https://issues.redhat.com/browse/HMS-4934   

Package groups is split at the handler and the doa, and given the amount of dao code i think that makes sense:    https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/package_groups.go   

So i wonder if we should maybe decide a better plan to divide these, implement it here and then add a card to refactor?  Thoughts?

Reply:
Alright, I've added module streams, it nows has it's own types/handler/dao/mocks. 

Let me know how it looks.